daemon/system-info.sh

```diff
@@ -8,111 +8,182 @@ KERNEL_VERSION="$(uname -r)"
-		fi
-		if [ -n "$(command -v lsb_release 2>/dev/null)" ]; then
-			if [ "${OS_DETECTION}" = "unknown" ]; then OS_DETECTION="lsb_release"; else OS_DETECTION="Mixed"; fi
-			if [ "${NAME}" = "unknown" ]; then NAME="$(lsb_release -is 2>/dev/null)"; fi
-			if [ "${VERSION}" = "unknown" ]; then VERSION="$(lsb_release -rs 2>/dev/null)"; fi
-			if [ "${ID}" = "unknown" ]; then ID="$(lsb_release -cs 2>/dev/null)"; fi
-		fi
-	fi
+    # Passed from outside - probably in docker run
+    VIRT_DETECTION="provided"
 fi
 
 # -------------------------------------------------------------------------------------------------
-# detect the virtualization
+# detect containers with heuristics
 
-VIRTUALIZATION="unknown"
-VIRT_DETECTION="none"
 CONTAINER="unknown"
 CONT_DETECTION="none"
 
-if [ -n "$(command -v systemd-detect-virt 2>/dev/null)" ]; then
-	VIRTUALIZATION="$(systemd-detect-virt -v)"
-	VIRT_DETECTION="systemd-detect-virt"
-	CONTAINER="$(systemd-detect-virt -c)"
-	CONT_DETECTION="systemd-detect-virt"
-else
-	if grep -q "^flags.*hypervisor" /proc/cpuinfo 2>/dev/null; then
-		VIRTUALIZATION="hypervisor"
-		VIRT_DETECTION="/proc/cpuinfo"
-	fi
+if [ "${CONTAINER}" = "unknown" ]; then
+        if [ -f /proc/1/sched ] ; then
+                IFS='(, ' read -r process _ </proc/1/sched
+                if [ "${process}" = "netdata" ]; then
+                        CONTAINER="container"
+                        CONT_DETECTION="process"
+                fi
+        fi
+        # ubuntu and debian supply /bin/running-in-container
+        # https://www.apt-browse.org/browse/ubuntu/trusty/main/i386/upstart/1.12.1-0ubuntu4/file/bin/running-in-container
+        if /bin/running-in-container >/dev/null 2>&1; then
+                CONTAINER="container"
+                CONT_DETECTION="/bin/running-in-container"
+        fi
+
+        # lxc sets environment variable 'container'
+        #shellcheck disable=SC2154
+        if [ -n "${container}" ]; then
+                CONTAINER="lxc"
+                CONT_DETECTION="containerenv"
+        fi
+
+        # docker creates /.dockerenv
+        # http://stackoverflow.com/a/25518345
+        if [ -f "/.dockerenv" ]; then
+                CONTAINER="docker"
+                CONT_DETECTION="dockerenv"
+        fi
+
 fi
 
 # -------------------------------------------------------------------------------------------------
-# detect containers with heuristics
+# detect the operating system
+
+# Initially assume all OS detection values are for a container, these are moved later if we are bare-metal
+
+CONTAINER_OS_DETECTION="unknown"
+CONTAINER_NAME="unknown"
+CONTAINER_VERSION="unknown"
+CONTAINER_VERSION_ID="unknown"
+CONTAINER_ID="unknown"
+CONTAINER_ID_LIKE="unknown"
+
+if [ "${KERNEL_NAME}" = "Darwin" ]; then
+        # Mac OS
+        OIFS="$IFS"
+        IFS=$'\n'
+        set $(sw_vers) > /dev/null
+        NAME=$(echo $1 | tr "\n\t" '  ' | sed -e 's/ProductName:[ ]*//' -e 's/[ ]*$//')
```

Comment:
```
In daemon/system-info.sh line 94:
        NAME=$(echo $1 | tr "\n\t" '  ' | sed -e 's/ProductName:[ ]*//' -e 's/[ ]*$//')
                    ^-- SC2086: Double quote to prevent globbing and word splitting.
```

Reply:
Cleaned up.